### PR TITLE
Change to LiteDB

### PR DESCRIPTION
**Pull request by cvium**
## Changes

Remove NanoApi
Use LiteDB
Cleanup

### CODE REVIEW

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs [hunk 1]
```diff
@@ -6,411 +6,201 @@
...
-                {
-                    instance = new DbRepo(dbPath);
-                }
-                return instance;
-            }
-        }
+        private readonly ILogger _logger;
+        private readonly IJsonSerializer _json;
 
+        public static DbRepo Instance;
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: shouldn't this be a public getter/private setter thing? or even better a proper singleton pattern?

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs [hunk 2]
```diff
@@ -6,411 +6,201 @@
...
-        //            {
-        //                result.Add(x.ItemId);
-        //            }
-        //        });
-        //        return result;
-        //    }
-        //}
-
-        public List<UserJson> GetUserInfos(long dtl, string userId, bool movies, bool tvshows, bool music, bool musicvideos, bool boxsets)
+        public List<UserJson> GetUserInfos(long dtl, string userId, IReadOnlyCollection<MediaType> filters)
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: why does `GetUserInfos` accept `filters`?

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs [hunk 3]
```diff
@@ -43,9 +44,9 @@ public Task RunAsync()
             return Task.CompletedTask;
         }
 
-        private bool FilterItem(BaseItem item, out int type)
+        private bool FilterItem(BaseItem item, out MediaType type)
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs]: this seems to be a more-or-less duplicate of `LibrarySyncNotification.FilterItem`, maybe extract common part somewhere?

File: Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj [hunk 4]
```diff
@@ -6,6 +6,10 @@
     <FileVersion>2.0.3.0</FileVersion>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <DebugType>full</DebugType>
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj]: umm, what's this?

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs (hunk 2 above)
Comment (cvium) [reply Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: It uses it?

File: Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj (hunk 4 above)
Comment (cvium) [reply Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj]: Just a failed attempt at making it build with deps. As I said, csproj is still a TODO ;)

File: Jellyfin.Plugin.KodiSyncQueue/Utils/Helpers.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Model.Entities;
+using MediaType = Jellyfin.Plugin.KodiSyncQueue.Entities.MediaType;
+
+namespace Jellyfin.Plugin.KodiSyncQueue.Utils
+{
+    public static class Helpers
+    {
+        public static bool FilterItem(BaseItem item, out MediaType type)
```
Comment (JustAMan) [inline Jellyfin.Plugin.KodiSyncQueue/Utils/Helpers.cs]: I suggest renaming it to better show what it does... maybe something like `GetItemType` or `FilterAndGetItemType`?

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs (hunk 2 above)
Comment (JustAMan) [reply Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: But why should it? Isn't it getting info on what users are available?..

Comment (cvium) [reply Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: Users can configure which libraries to sync in Kodi

Comment (JustAMan) [reply Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: I might not be understanding what this function does then...

**Pull request merged**